crates/engine/src/lib.rs

```diff
@@ -330,6 +335,21 @@ impl<T: Default> ExecutionContext<T> {
                 wasi_ctx.preopened_dir(Dir::open_ambient_dir(host, ambient_authority())?, guest)?;
         }
 
+        if component.core.wasm.enable_monitoring {
+            let sanitized_label = sanitize(&self.config.label);
+            let app_dir = match dirs::home_dir() {
+                Some(h) => h.join(SPIN_HOME).join(&sanitized_label),
+                None => PathBuf::from(&sanitized_label),
+            };
+
+            std::fs::create_dir_all(&app_dir)?;
```

Comment:
It would be nice to dedupe this w/ `save_output_to_logs` into an e.g. `fn ensure_app_dir(label: &str) -> Result<PathBuf>`.

I'd be fine doing that in a separate PR though.

Reply:
I can do that here, certainly. :)